enter.pollinations.ai/src/routes/admin.ts

```diff
@@ -0,0 +1,130 @@
+import { Hono } from "hono";
+import { getLogger } from "@logtape/logtape";
+import { drizzle } from "drizzle-orm/d1";
+import { eq } from "drizzle-orm";
+import { Polar } from "@polar-sh/sdk";
+import type { Env } from "../env.ts";
+import { user as userTable } from "../db/schema/better-auth.ts";
+import {
+    isValidTier,
+    syncUserTier,
+    getTierProductMap,
+    type TierName,
+} from "../tier-sync.ts";
+
+const log = getLogger(["hono", "admin"]);
+
+export const adminRoutes = new Hono<Env>()
+    .use("*", async (c, next) => {
+        // Use ENTER_TOKEN for admin authentication (already in GH secrets)
+        const adminKey = c.env.ENTER_TOKEN;
```

Comment:
I would prefer to use the admin plugin that is already installed to verify that the user has permissions (see https://www.better-auth.com/docs/plugins/admin). Manually verifying the token works, but i think its a bit error prone and would need to be duplicated in other routes.